### ISSUE TITLE
Missing "GADCustomNativeAdLoaderDelegate" Interface in Xamarin.Google.iOS.MobileAds

### ISSUE THREAD

**Issue by simspace-dr** [xamarin/GoogleApisForiOSComponents#536]
In the Xamarin.Google.iOS.MobileAds `ApiDefinition.cs` see that the `GADCustomNativeAd` and the `GADCustomNativeAdDelegate` interfaces appear to be fully implemented. However, there is no  `GADCustomNativeAdLoaderDelegate` interface defined. This presents a problem where we can use  `AdLoader` to request a custom native ad, but without a custom native ad loader delegate, we are not able to receive the ad being returned. Are there any plans to add the `GADCustomNativeAdLoaderDelegate` interface to the API definitions in the near future?

**Comment by Macjon** [comment]
@dmariogatto do you have an estimate when this will be released?

**Comment by dmariogatto** [comment]
@Macjon Nope, for the moment there isn't a roadmap, or regular cadence to this project so it's hard to predict.

**Comment by pboisso** [comment]
@dmariogatto, do you know why this project is not officially supported by Microsoft?  This is at the core of lots of iOS Apps.  MS really need to step up if they want to keep their Xamarin community. CC: @jfversluis

**Comment by dmariogatto** [comment]
@pboisso I wish I knew.

Feels like the important infrastructure around Xamarin isn't getting the required attention from MS (probably since they don't seem to use it for any of their major products).

We're starting to see Swift only APIs from Apple as well (e.g. StoreKit 2 in iOS 15) with no easy way to use them in .NET as the binding tools for Swift are not generally available.

If I were starting a new project today & had the requirement to use Firebase or a Swift Library/API, I would seriously be looking at other options.

Hopefully when MAUI is released that'll free up some resources, but at the moment my personal view is that the competitors are already on the podium with a bottle of champagne & Xamarin is stuck in the pit lane.

**Comment by jfversluis** [comment]
Thanks for the tag here! As you've guessed we've got a lot going on at the moment and it's hard to keep all the balls in the air. I also definitely hope that we have some more time to dedicate here or that we can at least be clear on what the status is for this project. I think some work is being done but I can't make any solid promises on timelines, sorry about that :(

**Comment by simspace-dev** [comment]
I just noticed in the API that the `CustomNativeAd` method to record clicks is named incorrectly ...
```
[Export ("performClickOnAssetWithKey:")]
[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
public virtual void RecordImpression (NSString assetKey);
```
It should be 
`public virtual void performClickOnAsset (NSString assetKey);`

See: https://developers.google.com/admob/ios/api/reference/Classes/GADCustomNativeAd?hl=en#-performclickonassetwithkey:

I will pull down the code and submit a PR for this tomorrow.

**Comment by simspace-dev** [comment]
Thanks for the comment @jfversluis!

**Comment by simspace-dr** [comment]
Does anyone have an idea if this is dead, or if the update to add the missing "GADCustomNativeAdLoaderDelegate" Interface will ever happen? Thanks!
Cc: @jfversluis, @dmariogatto
